network/infoblox.py

```diff
@@ -0,0 +1,380 @@
+        if m and m.group(1) == host:
+            self.invoke('delete', host_ref)
+            return "Object %s deleted" % host
+        else:
+            raise Exception("Received unexpected host reference: %s" % host_ref)
+
+    # ---------------------------------------------------------------------------
+    # set_extattr()
+    # ---------------------------------------------------------------------------
+    def set_extattr(self, object_ref,  attr_name, attr_value):
+        '''
+        Update the extra attribute value
+        '''
+        if not object_reft:
+            self.module.exit_json(msg="You must specify the option 'object_red'.")
+        payload = { "extattrs": { attr_name: { "value" : attr_value }}}
+        return self.invoke('put', object_ref, json=payload)
+
+
+# ---------------------------------------------------------------------------
+# MAIN
+# ---------------------------------------------------------------------------
+
+def main():
+    '''
+    Ansible module to manage infoblox opeartion by useing rest api
+    '''
+    module = AnsibleModule(
+        argument_spec=dict(
+            server      = dict(required=True),
+            username    = dict(required=True),
+            password    = dict(required=True),
+            action      = dict(required=True, choices=['get_host', 'get_network', 'get_next_available_ip', 'add_host','delete_host', 'set_extattr']),
+            host        = dict(required=False),
+            network     = dict(required=False, default=False),
+            address     = dict(required=False, default=False),
+            attr_name   = dict(required=False),
+            attr_value  = dict(required=False),
+            comment     = dict(required=False, default="Object managed by ansible-infoblox module"),
+            api_version = dict(required=False, default='1.7.1'),
+            dns_view    = dict(required=False, default='Private'),
+            net_view    = dict(required=False, default='default'),
+        ),
+        mutually_exclusive=[
+            ['network', 'address']
+            ],
+        required_together=[
+            ['attr_name','attr_value']
+            ],
+        supports_check_mode=True,
+    )
+
+    if not HAS_REQUESTS:
+         module.fail_json(msg="Library 'requests' is required. Use 'sudo pip install requests' to fix it.")
+
+    '''
+    Global vars
+    '''
+    server      = module.params["server"]
+    username    = module.params["username"]
+    password    = module.params["password"]
+    action      = module.params["action"]
+    host        = module.params["host"]
+    network     = module.params["network"]
+    address     = module.params["address"]
+    attr_name   = module.params["attr_name"]
+    attr_value  = module.params["attr_value"]
+    comment     = module.params["comment"]
+    api_version = module.params["api_version"]
+    dns_view    = module.params["dns_view"]
+    net_view    = module.params["net_view"]
+
+    try:
```

Comment:
I've removed the `try` termination dua a previous syntaxys error:

`except Exception as e:		
    module.fail_json(msg=str(e))`